daffodil-macro-lib/src/main/scala/org/apache/daffodil/lib/exceptions/AssertMacros.scala

```diff
@@ -28,7 +28,7 @@ object AssertMacros {
 
     q"""
     if (!($testAbortsIfFalse)) {
-         Assert.abort2("Usage error: " + $message, $testAsString)
+         throw new IllegalStateException("Usage error: " + $message)
```

Comment:
If you want this to use a standard java exception it should throw an Abort with IllegalArgumentException as the cause of the exception. Otherwise I fear this is going to break quite a lot of stuff. 

Changing the class being thrown is a big change. You need to scrutinize every catch in the code base to see if it has to change correspondingly.